test/e2e/suite/mixed_mode/mixed_mode_test.go

```diff
@@ -0,0 +1,408 @@
+				Expect(err).NotTo(HaveOccurred(), "creating client pod")
+				test.Logger.Info("Client pod created and ready", "name", clientPod.Name, "uid", clientPod.UID)
+
+				test.Logger.Info("Testing cross-VPC pod-to-pod connectivity (hybrid → cloud)")
+				err = kubernetes.TestPodToPodConnectivity(ctx, test.K8sClientConfig, test.K8sClient(), clientPod.Name, cloudPod.Name, "default", test.Logger)
+				Expect(err).NotTo(HaveOccurred(), "testing pod-to-pod connectivity")
+
+				test.Logger.Info("Cross-VPC pod-to-pod communication test completed successfully")
+			})
+
+			It("should enable cross-VPC pod-to-pod communication from cloud to hybrid nodes", func(ctx context.Context) {
+				hybridPod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "nginx-hybrid-reverse",
+						Namespace: "default",
+						Labels:    map[string]string{"app": "nginx-hybrid-reverse"},
+					},
+					Spec: corev1.PodSpec{
+						NodeSelector: hybridNodeSelector,
+						Containers: []corev1.Container{{
+							Name:    "nginx-hybrid-reverse",
+							Image:   "nginx:1.21",
+							Command: []string{"sh", "-c", "sed -i 's/listen.*80;/listen 8080;/' /etc/nginx/conf.d/default.conf && nginx -g 'daemon off;'"},
+							Ports:   []corev1.ContainerPort{{ContainerPort: 8080, Protocol: corev1.ProtocolTCP}},
+						}},
+					},
+				}
+				err := kubernetes.CreatePod(ctx, test.K8sClient(), hybridPod, test.Logger)
+				Expect(err).NotTo(HaveOccurred(), "creating hybrid pod")
+				test.Logger.Info("Hybrid pod created and ready", "name", hybridPod.Name, "uid", hybridPod.UID)
+
+				clientPod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-client-cloud",
+						Namespace: "default",
+						Labels:    map[string]string{"app": "test-client-cloud"},
+					},
+					Spec: corev1.PodSpec{
+						NodeSelector: cloudNodeSelector,
+						Containers: []corev1.Container{{
+							Name:    "test-client-cloud",
+							Image:   "curlimages/curl:7.85.0",
+							Command: []string{"sleep", "3600"},
+						}},
+					},
+				}
+				err = kubernetes.CreatePod(ctx, test.K8sClient(), clientPod, test.Logger)
+				Expect(err).NotTo(HaveOccurred(), "creating client pod")
+				test.Logger.Info("Cloud client pod created and ready", "name", clientPod.Name, "uid", clientPod.UID)
+
+				err = kubernetes.TestPodToPodConnectivity(ctx, test.K8sClientConfig, test.K8sClient(), clientPod.Name, hybridPod.Name, "default", test.Logger)
+				Expect(err).NotTo(HaveOccurred(), "testing pod-to-pod connectivity")
+
+				test.Logger.Info("Cross-VPC pod-to-pod communication test completed successfully")
+			})
+		})
+
+		Context("Cross-VPC Service Discovery", func() {
+			It("should enable cross-VPC service discovery from hybrid to cloud services", func(ctx context.Context) {
+				service, _, err := kubernetes.CreateServiceWithDeployment(ctx, test.K8sClient(), "nginx-service-cloud", "nginx:1.21", cloudNodeSelector, 80, 80, 1, test.Logger)
```

Comment:
The 1.21 version seems to be pretty old, ~3 years. Why are we using that version? Can we bump to newer versions? Or as suggested above, you could use the [image](https://github.com/aws/eks-hybrid/blob/main/test/e2e/kubernetes/pod.go#L44) from `CreateNginxPodInNode` method which gives you the latest one.